x-pack/plugins/enterprise_search/public/applications/index.tsx

```diff
@@ -6,28 +6,17 @@
 
 import React from 'react';
 import ReactDOM from 'react-dom';
-import { BrowserRouter, Route, Redirect } from 'react-router-dom';
+import { BrowserRouter } from 'react-router-dom';
 
 import { CoreStart, AppMountParams } from 'src/core/public';
 import { ClientConfigType } from '../plugin';
 
-import { AppSearch } from './app_search';
+import { App } from './app';
 
 export const renderApp = (core: CoreStart, params: AppMountParams, config: ClientConfigType) => {
   ReactDOM.render(
     <BrowserRouter basename={params.appBasePath}>
-      <Route exact path="/">
-        {/* This will eventually contain an Enterprise Search landing page,
-        and we'll also actually have a /workplace_search route */}
-        <Redirect to="/app_search" />
-      </Route>
-      <Route path="/app_search">
-        <AppSearch
-          http={core.http}
-          appSearchUrl={config.host}
-          setBreadcrumbs={core.chrome.setBreadcrumbs}
-        />
-      </Route>
+      <App core={core} config={config} />
```

Comment:
I moved this out to another file primarily so that I could:
- mount without `renderApp`, so I could have more control over testing
- Wrap this in MemoryRouter instead of Browser router in tests, so that I can test the routing

`applications.test.tsx` tests this entire thing using `renderApp`, so it's still covered.

Reply:
I think that's a good compromise, my only thought is - do we want to move `BrowserRouter` to app.tsx as well? From a RR point of view I think it makes sense to keep `<BrowserRouter>` and `<Route>` together 🤷

Reply:
I don't know what the *right* answer is, but I definitely like it here so that we can change out the router when we test.